src/Avalonia.Controls.TreeDataGrid/Primitives/TreeDataGridTemplateCell.cs

```diff
@@ -70,12 +110,20 @@ protected override void OnDataContextChanged(EventArgs e)
                 Content = cell.Value;
 
                 if (((ILogical)this).IsAttachedToLogicalTree)
+                {
                     ContentTemplate = cell.GetCellTemplate(this);
+                    EditingTemplate = cell.GetCellEditingTemplate?.Invoke(this);
+                }
             }
             else
             {
                 Content = null;
             }
         }
+
+        private void EditingContentPresenterLostFocus(object? sender, RoutedEventArgs e)
+        {
+            EndEdit();
```

Comment:
It won't work with cells with popups. Like combo box or custom controls. Like that customers app had.
DataGrid supports it in this way:
1. Handle lost focus
2. Check if new focused element is inside of popup and popup is descendent of the current cell (keep in mind, new focused control inside of the popup IS NOT direct descendent of the cell, neither in Avalonia nor UWP, as popup child has another root).
3. In which case subscribe on Popup.Closed or new child lost focus event.
4. End Edit only when new child lost focus occurred.

Let me find some source code...

Reply:
In general, everything related to "external editing element" from the data grid sources.

Reply:
Implemented that: note that I didn't find the reference to `Popup.Closed` in the source you linked so I didn't add code for that case.